### ISSUE TITLE
nextcloud Android Client crashes

### ISSUE THREAD

**Issue by CHHMA** [nextcloud/android#8560]
### Steps to reproduce
1.  Change pw on nextcloud web interface 
2.  Start nextcloud client on Android 
3.  When trying to connect, client crashes before requesting new credentials 

### Expected behaviour
- NC Android client starts and asks for new pw

### Actual behaviour
- NC Android client crashes before asking for new credentials 

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

### Environment data
************ CAUSE OF ERROR ************

java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter password
	at okhttp3.Credentials.basic(Unknown Source:8)
	at okhttp3.Credentials.basic$default(Credentials.kt:28)
	at okhttp3.Credentials.basic(Unknown Source:2)
	at com.owncloud.android.lib.common.OwnCloudClientFactory.createNextcloudClient(OwnCloudClientFactory.java:217)
	at com.owncloud.android.operations.RefreshFolderOperation.updatePredefinedStatus(RefreshFolderOperation.java:338)
	at com.owncloud.android.operations.RefreshFolderOperation.updateCapabilities(RefreshFolderOperation.java:311)
	at com.owncloud.android.operations.RefreshFolderOperation.updateOCVersion(RefreshFolderOperation.java:277)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:230)
	at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:359)
	at java.lang.Thread.run(Thread.java:929)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30160190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: HUAWEI
Device: HWELE
Model: ELE-L29
Id: HUAWEIELE-L29
Product: ELE-L29EEA

************ FIRMWARE ************
SDK: 29
Release: 10
Incremental: 10.1.0.150C431

Android version:

Device model: 

Stock or customized system:

Nextcloud app version:

Nextcloud server version:

Reverse proxy:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by signal4siju** [comment]
Hi Tobias,

I am facing this issue too. It happens post failed connection to server> I have 2 url updated on the client one is internal address and the other is external link using NOIP and opening the port 443 in my router. SSL is not configured. I observed this error when I tested the connection after closing the port 443 in my router.  I tried opening the app to check connection, at first it tried fetching the data, then client gave me a prompt of the certificate which used to appear during the installation. I accepted the certificate and then this error struck which is reported by the initial user . Post the error, while opening the App, it just takes me to the error page, it doesn't gives me an option to select the other url unfortunately.
